EU4toV2/Source/V2World/Province/Province.cpp

```diff
@@ -98,7 +99,8 @@ void V2::Province::addCore(const std::string& newCore)
 	}
 }
 
-void V2::Province::convertFromOldProvince(const std::vector<std::shared_ptr<EU4::Province>>& provinceSources,
+void V2::Province::convertFromOldProvince(const EU4::World& sourceWorld,
```

Comment:
please don't shove entire world into a v2:province, you're making it impossible to test. Just forward what it needs to know - is there it is there not the reform you're looking for.
For every province you'll be iterating over this. Have eu4::sourceworld prepare a bool or something and just pass that along.